core/controller/src/main/scala/whisk/core/controller/Authenticate.scala

```diff
@@ -79,4 +74,25 @@ trait Authenticate {
             Future.successful(None)
         }
     }
+
+  def validateCertificate(name: Option[String])(implicit transid: TransactionId): Future[Option[Identity]] = {
+      name flatMap { entityName =>
+          val name = entityName.substring("CN=".length, entityName.length)
+          Try {
+              //It also supports getting Identity by subject
+              val future = Identity.get(authStore, EntityName(name)) map { result =>
```

Comment:
I think this is only going to work for the case where the subject has a single namespace and both values match.
If the CN name is going to be a namespace - rather than a subject id, this will generally not work. In retrospect, it should have been obvious to me CN name should be a namespace. Do you agree or disagree?